samples/bluetooth/mesh/light_switch/src/model_handler.c

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+/**
+ * @file
+ * @brief Model handler for the light switch.
+ *
+ * Instantiates a Generic OnOff Client model for each button on the devkit, as
+ * well as the standard foundation models. Handles all application behavior
+ * related to the models.
+ */
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/mesh/models.h>
+#include <dk_buttons_and_leds.h>
+#include "model_handler.h"
+
+/* Light switch behavior */
+
+/** Context for a single light switch. */
+struct button {
+	/** Current light status of the corresponding server. */
+	bool status;
+	/** Generic OnOff client instance for this switch. */
+	struct bt_mesh_onoff_cli client;
+};
+
+static void status_handler(struct bt_mesh_onoff_cli *cli,
+			   struct bt_mesh_msg_ctx *ctx,
+			   const struct bt_mesh_onoff_status *status);
+
+static struct button buttons[4] = {
+	[0 ... 3] = { .client = BT_MESH_ONOFF_CLI_INIT(&status_handler) },
+};
+
+static void status_handler(struct bt_mesh_onoff_cli *cli,
+			   struct bt_mesh_msg_ctx *ctx,
+			   const struct bt_mesh_onoff_status *status)
+{
+	struct button *button =
+		CONTAINER_OF(cli, struct button, client);
+	int index = button - &buttons[0];
+
+	button->status = status->present_on_off;
+	dk_set_led(index, status->present_on_off);
+
+	printk("Button %d: Received response: %s\n", index + 1,
+	       status->present_on_off ? "on" : "off");
+}
+
+static void button_handler_cb(u32_t pressed, u32_t changed)
+{
+	if (!bt_mesh_is_provisioned()) {
+		return;
+	}
+
+	for (int i = 0; i < 4; ++i) {
+		if (!(pressed & changed & BIT(i))) {
+			continue;
+		}
+
+		struct bt_mesh_onoff_set set = {
+			.on_off = !buttons[i].status,
+		};
+		int err;
+
+		/* As we can't know how many nodes are in a group, it doesn't
+		 * make sense to send acknowledged messages to group addresses -
+		 * we won't be able to make use of the responses anyway.
+		 */
+		if (bt_mesh_model_pub_is_unicast(buttons[i].client.model)) {
+			err = bt_mesh_onoff_cli_set(&buttons[i].client, NULL,
+						    &set, NULL);
+		} else {
+			err = bt_mesh_onoff_cli_set_unack(&buttons[i].client,
+							  NULL, &set);
+			if (!err) {
+				/* There'll be no response status for the
+				 * unacked message. Set the state immediately.
+				 */
+				buttons[i].status = set.on_off;
+				dk_set_led(i, set.on_off);
+			}
+		}
+
+		if (err) {
+			printk("OnOff %d set failed: %d\n", i + 1, err);
+		}
+	}
+}
+
+/** Configuration server definition */
+static struct bt_mesh_cfg_srv cfg_srv = {
+	.relay = IS_ENABLED(CONFIG_BT_MESH_RELAY),
+	.beacon = BT_MESH_BEACON_ENABLED,
+	.frnd = IS_ENABLED(CONFIG_BT_MESH_FRIEND),
+	.gatt_proxy = IS_ENABLED(CONFIG_BT_MESH_GATT_PROXY),
+	.default_ttl = 7,
```

Comment:
Is this 7 something we could put in a header somewhere as recommended, or default?